lua/diffview/vcs/adapters/hg/init.lua

```diff
@@ -1,18 +1,1141 @@
+      "status",
+      "--print0",
+      "--unknown",
+      "--no-status",
+      "--template={path}\\n"
+    ),
+    cwd = self.ctx.toplevel,
+    ---@type Job
+    on_exit = function(j)
+      utils.handle_job(j, {
+        debug_opt = {
+          context = "HgAdapter>untracked_files()",
+          func = "s_debug",
+          debug_level = 1,
+          no_stdout = true,
+        },
+      })
+
+      if j.code ~= 0 then
+        callback(j:stderr_result() or {}, nil)
+        return
+      end
+
+      local files = {}
+      for _, s in ipairs(j:result()) do
+        table.insert(
+          files,
+          FileEntry.with_layout(opt.default_layout, {
+            adapter = self,
+            path = s,
+            status = "?",
+            kind = "working",
+            revs = {
+              a = left,
+              b = right,
+            }
+          })
+        )
+      end
+      callback(nil, files)
+    end,
+  }):start()
+end, 5)
+
+---@param self HgAdapter
+---@param path string
+---@param rev? Rev
+---@param callback fun(stderr: string[]?, stdout: string[]?)
+HgAdapter.show = async.wrap(function(self, path, rev, callback)
+  -- File did not exist, need to return an empty buffer
+  if not(rev) or (rev:object_name() == self.Rev.NULL_TREE_SHA) then
+    callback(nil, {})
+    return
+  end
+
+  local job = Job:new({
+    command = self:bin(),
+    args = self:get_show_args(path, rev),
+    cwd = self.ctx.toplevel,
+    ---@type Job
+    on_exit = async.void(function(j)
+      local context = "HgAdapter.show()"
+      utils.handle_job(j, {
+        fail_on_empty = true,
+        context = context,
+        debug_opt = { no_stdout = true, context = context },
+      })
+
+      if j.code ~= 0 then
+        callback(j:stderr_result() or {}, nil)
+        return
+      end
+
+      local out_status
+
+      if #j:result() == 0 then
+        async.util.scheduler()
+        out_status = vcs_utils.ensure_output(2, { j }, context)
+      end
+
+      if out_status == JobStatus.ERROR then
+        callback(j:stderr_result() or {}, nil)
+        return
+      end
+
+      callback(nil, j:result())
+    end),
+  })
+  -- Problem: Running multiple 'show' jobs simultaneously may cause them to fail
+  -- silently.
+  -- Solution: queue them and run them one after another.
+  vcs_utils.queue_sync_job(job)
+end, 4)
+
+HgAdapter.flags = {
+  ---@type FlagOption[]
+  switches = {
+    { '-f', '--follow', 'Follow renames' },
+    { '-M', '--no-merges', 'List no merge changesets' },
+  },
+  ---@type FlagOption[]
+  options = {
+    { '=r', '--rev=', 'Revspec', prompt_label = "(Revspec)" },
+    { '=l', '--limit=', 'Limit the number of changesets' },
+    { '=u', '--user=', 'Filter on user' },
+    { '=k', '--keyword=', 'Filter by keyword' },
+    { '=b', '--branch=', 'Filter by branch' },
+    { '=B', '--bookmark=', 'Filter by bookmark' },
+    { '=I', '--include=', 'Include files' },
+    { '=E', '--exclude=', 'Exclude files' },
+  },
+}
```

Comment:
I just recently changed how these flag options are declared, sorry about that 😅 Take a look in the git adapter:
https://github.com/zegervdv/diffview.nvim/blob/a281b78204a5b3d2b54ff395df094824fac7b09e/lua/diffview/vcs/adapters/git/init.lua#L1656

Reply:
No worries, looks like a straightforward change